protocols/request-response/src/handler.rs

```diff
@@ -79,6 +80,8 @@ where
     inbound_request_id: Arc<AtomicU64>,
 
     worker_streams: futures_bounded::FuturesMap<RequestId, Result<Event<TCodec>, io::Error>>,
+
+    observed_addr: Arc<Multiaddr>,
```

Comment:
Any reason why use a `Arc` for the `Multiaddr`? The multiaddr does uses `Arc` internally.

Reply:
I have removed it.